signer/src/stacks/api.rs

```diff
@@ -590,6 +650,42 @@ fn check_result<T>(client: &mut StacksClient, result: Result<T, Error>) -> Resul
 }
 
 impl StacksInteract for StacksClient {
+    async fn get_current_signer_set(
+        &mut self,
+        contract_principal: &StacksAddress,
+    ) -> Result<Vec<PublicKey>, Error> {
+        use clarity::vm::types::{BuffData, ListData, SequenceData, Value};
+
+        let result = check_result(
+            self,
+            StacksClient::get_data_var(
+                self,
+                contract_principal,
+                &ContractName::from("sbtc-registry"),
+                &ClarityName::from("current-signer-set"),
+            )
+            .await,
+        )?;
```

Comment:
I find it a lot easier to read if this is broken out into two different variables, like:
```rust
let res = self.get_data_var(
    contract_principal,
    &ContractName::from("sbtc-registry"),
    &ClarityName::from("current-signer-set"),
)
.await;
let result = check_result(self, res)?;
```

Reply:
Separated them.

Tbh `check_result` feels a bit weird here, feels more like something that should be refactored up into the `StacksClient` itself.

Reply:
Yeah, I can see that.

Maybe it's just me, but I feel like the `StacksClient` should have simple functions that hit the endpoint and do basic conversions into the return type. For additional functionality like retries and `check_result`, they should be done at a layer on top of these simple functions. Well, that's why I pushed to move them outside of the core functions themselves, with the compromise being that they are now in the traits. But yeah, it looks a bit weird to me too.

Reply:
I think that since `check_result` right now is only checking for connection-related errors and rotating the endpoint (which belongs to `StacksClient`) that that feels more "transport-level" i.e. `StacksClient`. Ideally I think that `StacksInteract` should be oblivious to such things - either it succeeded or failed. `StacksClient` could have configurable timeout/retry policies etc. which would give the caller more control over when it should error.

My $0.02 :)